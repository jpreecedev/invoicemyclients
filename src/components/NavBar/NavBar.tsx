import React from 'react'
import { Link } from 'react-router-dom'

import { siteConfig } from '../../site-config'
import { ROUTES } from '../../routes'
import { SignUpButton } from '../SignUpButton'

const NavBar: React.FC = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-light">
      <div className="container">
        <Link className="navbar-brand" to={ROUTES.home}>
          {siteConfig.title}
        </Link>

        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarCollapse"
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className="collapse navbar-collapse" id="navbarCollapse">
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarCollapse"
            aria-controls="navbarCollapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <i className="fe fe-x"></i>
          </button>

          <ul className="navbar-nav ml-auto">
            <li className="nav-item dropdown">
              <Link className="nav-link" to={ROUTES.nav.dashboard}>
                Dashboard
              </Link>
            </li>
            <li className="nav-item dropdown">
              <Link className="nav-link" to={ROUTES.nav.clients}>
                Clients
              </Link>
            </li>
            <li className="nav-item dropdown">
              <Link className="nav-link" to={ROUTES.nav.invoices}>
                Invoices
              </Link>
            </li>
            <li className="nav-item dropdown">
              <Link className="nav-link" to={ROUTES.nav.nagging}>
                Nagging
              </Link>
            </li>
            <li className="nav-item dropdown">
              <Link className="nav-link" to={ROUTES.nav.support}>
                Support
              </Link>
            </li>
          </ul>

          <SignUpButton />
        </div>
      </div>
    </nav>
  )
}

export { NavBar }
