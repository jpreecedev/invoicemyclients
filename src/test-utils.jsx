import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
