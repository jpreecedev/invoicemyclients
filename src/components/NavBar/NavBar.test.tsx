import React from 'react'

import { renderWithRouter } from '../../test-utils'
import { NavBar } from './NavBar'

describe('<NavBar /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<NavBar />).container).toMatchSnapshot()
  })
})
