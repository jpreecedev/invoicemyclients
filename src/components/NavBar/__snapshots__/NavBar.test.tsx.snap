// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
</div>
`;
